File: lesunny/058_기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int length = progresses.length;
        int[] elapsedDay = new int[length];
        ArrayList<Integer> ansAL = new ArrayList<>();
        int[] answer;
        
        for(int i=0; i<length; i++){
            int num = (100-progresses[i])/speeds[i];
            if((double)(100-progresses[i])/speeds[i] > (double)num)
                elapsedDay[i] = (100-progresses[i])/speeds[i] + 1;
            else
                elapsedDay[i] = (100-progresses[i])/speeds[i];
        }
        
        for(int i=0; i<length; i++){
            int cnt = 1;
            
            for(int j=i+1; j<length ; j++){
                if(elapsedDay[i] < elapsedDay[j] ){ 
                    i=j-1;
                    break;
                }else{
                    cnt++;
                }
                
                if(j==length-1)
                    i=j;
            }                    
            
            ansAL.add(cnt);
        }
        
        answer = new int[ansAL.size()];
        int idx=0;
        for(int ans : ansAL){
            answer[idx++] = ans;
        }
                
        return answer;
    }
}

File: JIWON/058 기능개발.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> a;
    double temp_pro;
    
    for(int i=0;i<progresses.size();i++){
        temp_pro = ceil((double)(100 - progresses[i]) / (double)speeds[i]);
        answer.push_back(temp_pro);
    }
    
    for(int i=0;i<answer.size()-1;i++){
        for(int j=i+1;j<answer.size();j++){
            if(answer[i] > answer[j]){
                answer[j] = answer[i];
            }
        }
    }
    
    int count = 1;
    
    for(int i=0;i<answer.size();i++){
        if(answer[i] == answer[i+1]){
            count++;
        } else{
            a.push_back(count);
            
            count = 1;
        }
    }
    
    return a;
}
